ui/plugin-system/src/context/TimeRangeStateProvider.tsx

```diff
@@ -0,0 +1,59 @@
+// Copyright 2021 The Perses Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import React, { useState, useMemo, useCallback } from 'react';
+import { TimeRangeValue, AbsoluteTimeRange, toAbsoluteTimeRange, isRelativeTimeRange } from '@perses-dev/core';
+import { TimeRange, TimeRangeContext } from '../runtime/time-range';
+
+export interface TimeRangeProviderProps {
+  initialValue: TimeRangeValue;
+  children?: React.ReactNode;
+  onDateRangeChange?: (e: TimeRangeValue) => void;
```

Comment:
nit: Since the type is called `TimeRange`, is it better for this to be called `onTimeRangeChange` ?

Reply:
yeah good call, this is updated now